modules/custom-status/lib/components/custom-status-sidebar.js

```diff
@@ -0,0 +1,84 @@
+import { Button, SelectControl } from '@wordpress/components';
+import { dispatch, select, subscribe } from '@wordpress/data';
+import { PluginPostStatusInfo } from '@wordpress/edit-post';
+import { store as editorStore } from '@wordpress/editor';
+import { useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+import { isUsingWorkflowStatus } from '../util';
+
+/**
+ * Map Custom Statuses as options for SelectControl
+ */
+const statusOptions = VW_CUSTOM_STATUSES.status_terms.map( customStatus => ( {
+	label: customStatus.name,
+	value: customStatus.slug,
+} ) );
+
+/**
+ * Subscribe to changes so we can set a default status.
+ */
+subscribe( function () {
+	const postId = select( editorStore ).getCurrentPostId();
+	if ( ! postId ) {
+		// Post isn't ready yet so don't do anything.
+		return;
+	}
+
+	// For new posts, we need to force the default custom status which is the first entry.
+	const isCleanNewPost = select( editorStore ).isCleanNewPost();
+
+	if ( isCleanNewPost ) {
+		dispatch( editorStore ).editPost( {
+			status: statusOptions[ 0 ].value,
+		} );
+	}
+} );
+
+/**
+ * Custom status component
+ * @param object props
+ */
+export default function CustomStatusSidebar( { onUpdateStatus, postType, status } ) {
+	const [ isEditingStatus, setIsEditingStatus ] = useState( false );
+
+	const handleChangeStatus = statusSlug => {
+		onUpdateStatus( statusSlug );
+		dispatch( editorStore ).savePost();
+		setIsEditingStatus( false );
+	};
+
+	if ( ! isUsingWorkflowStatus( postType, status ) ) {
+		return null;
+	}
+
+	const customStatusName = VW_CUSTOM_STATUSES.status_terms.find( t => t.slug === status )?.name;
+
+	return (
+		<PluginPostStatusInfo
+			className={ `vip-workflow-extended-post-status vip-workflow-extended-post-status-${ status }` }
+		>
+			<h4>{ __( 'Extended Post Status', 'vip-workflow' ) }</h4>
```

Comment:
> I'm thinking that we be bold about this, and take this out entirely. So the custom save button becomes the best way to move a post through the various statuses that are supported.

I agree with this approach but it also makes me wonder if we should support the reverse direction as well? Perhaps only certain users like editors? I'm thinking that there would certainly be certain scenarios when an editor might might want to make a post back to a previous status. E.g.: An `assigned` post no longer has an author working on it so the editor decides to move it back to `pitch`

Reply:
> I'm thinking that we be bold about this, and take this out entirely.

There is one main problem I'm concerned with, and that's accidentally moving a status forward. An editor might hit the blue toolbar button expecting it to be a "Save", and then accidentally move it to a status it isn't ready for yet. If that happens, the only "fix" is to recreate the post, or just ignore the custom status.

I don't have a great solution for that, but maybe we could make the UI more friendly for regular saves. Here's a two-step concept where we just have a generic "Save" button that displays options for promote/save draft after clicking:

![save-2-step](https://github.com/user-attachments/assets/10122277-588a-4909-a80d-4f4223794779)

Reply:
I like @hanifn's idea to have a permissions-based control, like maybe only showing the "Edit" button for `manage_options` users. However, we don't have any backend controls for only allowing some users to move statuses backward, only a guard against publishing. How about we leave the "Edit" button in for now and punt the work to allow higher-level editing for another PR?

Reply:
My main concern was how confusing it could be to have both of those in there. I'm good with leaving it for now, and thinking about it as part of the approve/reject flow. We can potentially do the exact thing that you are suggesting since we do have a full sidebar UI for ourselves.

Could you add a short description to that extended status explaining this briefly? That way it's a touch less confusing.

Reply:
I added some help text below the "Edit" button to give some context:

![extended-post-status-help](https://github.com/user-attachments/assets/24317d9c-2cfb-40aa-bbcd-19d77bc87b7b)